frontend/src/metabase/query_builder/components/filters/pickers/TextPicker.jsx

```diff
@@ -42,7 +42,10 @@ export default class TextPicker extends Component {
             <div>
                 <ul>
                     {values.map((value, index) =>
-                        <li className="FilterInput px1 pt1 relative">
+                        <li
+                            className="FilterInput px1 pt1 relative"
+                            key={index}
```

Comment:
Did you want me to change this from index here? 

I'll go ahead and create a separate issue for the behavior you noted.